packages/payload/src/graphql/schema/buildWhereInputType.ts

```diff
@@ -89,26 +89,36 @@ const buildWhereInputType = ({
 
   const fieldName = formatName(name)
 
+  const recursiveFields = {
+    AND: {
+      type: new GraphQLList(
+        new GraphQLInputObjectType({
+          name: `${fieldName}_where_and`,
+          fields: () => ({
+            ...fieldTypes,
+            ...recursiveFields,
+          }),
+        }),
+      ),
+    },
+    OR: {
+      type: new GraphQLList(
+        new GraphQLInputObjectType({
+          name: `${fieldName}_where_or`,
+          fields: () => ({
+            ...fieldTypes,
+            ...recursiveFields,
```

Comment:
I am not sure how this is working? It is spreading an object that is not yet created, right?

I am not sure that this fix is the _right_ fix. Feels like we need actual recursion here. I might be able to take this one off your plate, but I have to understand the problem better first.

Reply:
Because `fields` is a callable function, I see how this works syntax-wise, but I'm not sure I understand how this change affects the actual GraphQL output. That is why I was looking through the generated schema.graphql files and it seems that it doesn't do anything. I think it is a good idea somebody to better understand what is supposed to be happening here and have a clear example of it being able to do what it is supposed to. 

Yesterday I played around with the GraphQL playground trying to understand, but I'm still not sure what this solves.

Reply:
@DanRibbens Because of the way we were originally generating graphql schemas, a query like this one [here](https://payloadcms.com/docs/queries/overview#and-or-logic) - was not possible in graphql.

The reason for it was because the types `AND` & `OR` were not being set to the `where_or` & `where_and` inputs of a collection type

As an example:

```
input Collection1_where_and {
  testing: Collection1_testing_operator
  title: Collection1_title_operator
  meta__title: Collection1_meta__title_operator
  updatedAt: Collection1_updatedAt_operator
  createdAt: Collection1_createdAt_operator
  id: Collection1_id_operator
  AND: [Collection1_where_and]
  OR: [Collection1_where_or]
}

input Collection1_where_or {
  testing: Collection1_testing_operator
  title: Collection1_title_operator
  meta__title: Collection1_meta__title_operator
  updatedAt: Collection1_updatedAt_operator
  createdAt: Collection1_createdAt_operator
  id: Collection1_id_operator
  AND: [Collection1_where_and]
  OR: [Collection1_where_or]
}
```

Prior to the change these input types were not getting AND & OR - so in graphql you wouldn't be able to nest these operators inside of each other - which should be possible.

Reply:
@DanRibbens The issue corresponding to this PR is this one: https://github.com/payloadcms/payload/issues/3538

It does a good job explaining the problem.

Reply:
Yeah I see now. 

I think this looks like a good change? This would be easy to show a before / after of the playground if that is what this solves. @PatrikKozak can you recreate the issue and plop two screenshots above to show the broken and working playgrounds?

Reply:
@JarrodMFlesch Just added two screenshots (`before` & `after`) in the original description!